src/lxc/confile.c

```diff
@@ -403,7 +403,8 @@ static int get_network_netdev_idx(const char *key)
 	int ret, idx;
 
 	if (*key < '0' || *key > '9')
-		return -1;
+		return 0; //it is of the form lxc.network.link
```

Comment:
You probably want a more specific test, because your change also makes "lxc.network.a" a valid string, which it isn't.